### PR TITLE
Adding Linters for Checking for Deadcode

### PR DESCRIPTION
**Pull request by danielhelfand**
This pull request adds linters to detect if any deadcode is in any changes being added to kapp. Currently, there is very little deadcode project, but might be helpful to have a way to protect against it moving forward.

### CODE REVIEW

File: pkg/kapp/logger/ui.go [hunk 1]
```diff
@@ -11,7 +11,9 @@ import (
 )
 
 const (
+	// nolint: deadcode,unused
```
Comment (cppforlife) [inline pkg/kapp/logger/ui.go]: i think we should use these below in Info and Error methods.

File: pkg/kapp/resources/association_label.go [hunk 2]
```diff
@@ -12,7 +12,6 @@ import (
 
 const (
 	kappAssociationLabelKey = "kapp.k14s.io/association"
-	kappAssociationLabelV1  = "v1"
```
Comment (cppforlife) [inline pkg/kapp/resources/association_label.go]: this one is actually looks like should be used in v1Value() of AssociationLabel. it's currently using kappIdentityAnnV1 which has a same value but technically isnt for this class.

Comment (danielhelfand) [reply pkg/kapp/resources/association_label.go]: Added back and now using `kappAssociationLabelV1`

File: pkg/kapp/logger/ui.go (hunk 1 above)
Comment (danielhelfand) [reply pkg/kapp/logger/ui.go]: Using in associated methods now.

**Pull request merged**